pkg/dataobj/consumer/partition_processor.go

```diff
@@ -54,14 +58,22 @@ func newPartitionProcessor(ctx context.Context, client *kgo.Client, builderCfg d
 	}
 	reg = prometheus.WrapRegistererWith(prometheus.Labels{
 		"partition": strconv.Itoa(int(partition)),
+		"tenant":    tenantID,
 	}, reg)
 
 	metrics := newPartitionOffsetMetrics()
 	if err := metrics.register(reg); err != nil {
 		level.Error(logger).Log("msg", "failed to register partition metrics", "err", err)
 	}
 
-	metastoreManager, err := metastore.NewMetastoreManager(bucket, tenantID, logger, reg)
+	uploader := uploader.New(uploader.UploaderConfig{
```

Comment:
I'm not sure whether to instantiate these here or lazily inside the initBuilder function. We use initBuilder to avoid memory usage when the processor isn't doing anything, but both the Uploader & metastore manager don't use much memory on their own. It might be nicer to keep everything together though?
Adding to that, the metastore manager has its own lazy init function, which could potentially be removed if we do it here instead.